samples/otlptracegrpc/app.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+/*app.ts*/
+import { trace, diag, DiagConsoleLogger, DiagLogLevel } from '@opentelemetry/api';
+import express, { Express, Request, Response } from 'express';
+import { rollTheDice } from './dice';
+
+import { NodeSDK } from '@opentelemetry/sdk-node';
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-grpc';
+import { AuthClient, GoogleAuth } from 'google-auth-library';
+import { credentials } from '@grpc/grpc-js';
+
+const tracer = trace.getTracer('dice-server', '0.1.0');
+
+const PORT: number = parseInt(process.env.PORT || '8080');
+const app: Express = express();
+
+async function main() {
+  diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.DEBUG);
+
+  async function getAuthenticatedClient(): Promise<AuthClient> {  
+    const auth: GoogleAuth = new GoogleAuth({
+      scopes: 'https://www.googleapis.com/auth/cloud-platform',
+    });
+    const client: AuthClient = await auth.getClient();
+    return client;    
+  }
+  const authenticatedClient: AuthClient = await getAuthenticatedClient();
```

Comment:
I think having this in a separate one-time function is hurting readability vs inline

Reply:
I think having an inline function is hurting readability, but the function itself is well-defined. I will move the inline function outside main.